src/subdatatable/subdatatable.jl

```diff
@@ -94,36 +94,31 @@ function SubDataTable(parent::DataTable, row::Integer)
 end
 
 function SubDataTable{S <: Integer}(parent::DataTable, rows::AbstractVector{S})
-    return view(parent, Int(rows))
+    return SubDataTable(parent, Int(rows))
```

Comment:
Thanks, Christof! You're right, `Int(rows)` doesn't work. I think I'll change this to `convert(Vector{Int}, rows)` instead as that seems more efficient. I'll add tests for these too.

```julia
julia> x = convert(Vector{Integer}, rand(1:100, 10000));

julia> @benchmark Int.(x)
BenchmarkTools.Trial:
  memory estimate:  99.89 KiB
  allocs estimate:  640
  --------------
  minimum time:     1.146 ms (0.00% GC)
  median time:      1.227 ms (0.00% GC)
  mean time:        1.307 ms (0.67% GC)
  maximum time:     4.195 ms (69.78% GC)
  --------------
  samples:          3739
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> @benchmark convert(Vector{Int}, x)
BenchmarkTools.Trial:
  memory estimate:  78.20 KiB
  allocs estimate:  2
  --------------
  minimum time:     632.115 μs (0.00% GC)
  median time:      666.270 μs (0.00% GC)
  mean time:        708.323 μs (1.05% GC)
  maximum time:     3.996 ms (81.92% GC)
  --------------
  samples:          7013
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> x = convert(Vector{UInt}, rand(1:100, 10000));

julia> @benchmark Int.(x)
BenchmarkTools.Trial:
  memory estimate:  99.23 KiB
  allocs estimate:  621
  --------------
  minimum time:     502.599 μs (0.00% GC)
  median time:      526.251 μs (0.00% GC)
  mean time:        571.313 μs (1.85% GC)
  maximum time:     3.742 ms (84.36% GC)
  --------------
  samples:          8348
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> @benchmark convert(Vector{Int}, x)
BenchmarkTools.Trial:
  memory estimate:  78.20 KiB
  allocs estimate:  2
  --------------
  minimum time:     6.188 μs (0.00% GC)
  median time:      9.686 μs (0.00% GC)
  mean time:        19.369 μs (30.94% GC)
  maximum time:     532.072 μs (95.30% GC)
  --------------
  samples:          10000
  evals/sample:     5
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> x = convert(Vector{BigInt}, rand(1:100, 10000));

julia> @benchmark Int.(x)
BenchmarkTools.Trial:
  memory estimate:  99.23 KiB
  allocs estimate:  621
  --------------
  minimum time:     581.778 μs (0.00% GC)
  median time:      618.722 μs (0.00% GC)
  mean time:        660.770 μs (1.75% GC)
  maximum time:     4.633 ms (79.66% GC)
  --------------
  samples:          7243
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> @benchmark convert(Vector{Int}, x)
BenchmarkTools.Trial:
  memory estimate:  78.20 KiB
  allocs estimate:  2
  --------------
  minimum time:     79.756 μs (0.00% GC)
  median time:      83.737 μs (0.00% GC)
  mean time:        105.017 μs (8.60% GC)
  maximum time:     3.753 ms (93.77% GC)
  --------------
  samples:          10000
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> x = rand(1:100, 10000);

julia> @benchmark Int.(x)
BenchmarkTools.Trial:
  memory estimate:  99.23 KiB
  allocs estimate:  621
  --------------
  minimum time:     503.096 μs (0.00% GC)
  median time:      523.553 μs (0.00% GC)
  mean time:        567.104 μs (2.14% GC)
  maximum time:     3.929 ms (83.60% GC)
  --------------
  samples:          8415
  evals/sample:     1
  time tolerance:   5.00%
  memory tolerance: 1.00%

julia> @benchmark convert(Vector{Int}, x)
BenchmarkTools.Trial:
  memory estimate:  0 bytes
  allocs estimate:  0
  --------------
  minimum time:     60.784 ns (0.00% GC)
  median time:      64.697 ns (0.00% GC)
  mean time:        69.840 ns (0.00% GC)
  maximum time:     221.769 ns (0.00% GC)
  --------------
  samples:          10000
  evals/sample:     980
  time tolerance:   5.00%
  memory tolerance: 1.00%
```